terralab/logic/submit_logic.py

```diff
@@ -0,0 +1,95 @@
+# logic/submit_logic.py
+
+import logging
+import uuid
+from teaspoons_client import (
+    PipelineRunsApi,
+    PreparePipelineRunResponse,
+    PreparePipelineRunRequestBody,
+    StartPipelineRunRequestBody,
+    JobControl,
+)
+
+from terralab.logic.upload_download_logic import upload_file_with_signed_url
+from terralab.client import ClientWrapper
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+## API wrapper functions
+SIGNED_URL_KEY = "signedUrl"
+
+
+def prepare_pipeline_run(
+    pipeline_name: str, job_id: uuid.UUID, pipeline_version: int, pipeline_inputs: dict
+) -> dict:
+    """Call the preparePipelineRun Teaspoons endpoint.
+    Return a dictionary of {input_name: signed_url}."""
+    prepare_pipeline_run_request_body: PreparePipelineRunRequestBody = (
+        PreparePipelineRunRequestBody(
+            jobId=job_id,
+            pipelineVersion=pipeline_version,
+            pipelineInputs=pipeline_inputs,
+        )
+    )
+
+    with ClientWrapper() as api_client:
+        pipeline_runs_client = PipelineRunsApi(api_client=api_client)
+        response: PreparePipelineRunResponse = (
+            pipeline_runs_client.prepare_pipeline_run(
+                pipeline_name, prepare_pipeline_run_request_body
+            )
+        )
+
+        result = response.file_input_upload_urls
+        return {
+            input_name: signed_url_dict.get(SIGNED_URL_KEY)
+            for input_name, signed_url_dict in result.items()
+        }
+
+
+def start_pipeline_run(
+    pipeline_name: str, job_id: uuid.UUID, description: str
+) -> uuid.UUID:
+    """Call the startPipelineRun Teaspoons endpoint and return the Async Job Response."""
+    start_pipeline_run_request_body: StartPipelineRunRequestBody = (
+        StartPipelineRunRequestBody(
+            description=description, jobControl=JobControl(id=job_id)
+        )
+    )
+    with ClientWrapper() as api_client:
+        pipeline_runs_client = PipelineRunsApi(api_client=api_client)
+        return pipeline_runs_client.start_pipeline_run(
+            pipeline_name, start_pipeline_run_request_body
+        ).job_report.id
+
+
+## submit action
+
+
+def prepare_upload_start_pipeline_run(
+    pipeline_name: str, pipeline_version: str, pipeline_inputs: dict, description: str
+) -> uuid.UUID:
+    """Prepare pipeline run, upload input files, and start pipeline run.
+    Returns the uuid of the job."""
+    # generate a job id for the user
+    job_id = str(uuid.uuid4())
+    LOGGER.debug(f"Generated job_id {job_id}")
```

Comment:
do we want to present this value to the user in case somethign fails between prepare and start so they know what job id they should try to debug?

Reply:
yeah i had this as info and changed it to debug, but can add it back for this reason